wellcomeml/ml/keras_vectorizer.py

```diff
@@ -38,6 +39,32 @@ def __init__(self, vocab_size=None, sequence_length=None, oov_token="<OOV>",
         self.sequence_length = sequence_length
         self.tokenizer_library = tokenizer_library
 
+    def _infer_from_data(self, X, load_buffer=1000):
+        # We could look at a sample for more efficient
+        max_sequence_length = 1
+
+        def update_max_sequence_length(X_buffer, max_sequence_length, load_buffer):
+            X_vec = self.transform(X_buffer)
```

Comment:
If time is a big problem with this, then I wonder if there is something you can do to not have to transform every `X_buffer` (I assume this is the costly bit). 
For example, if `sum([len(x) for x in X_buffer]) < max_sequence_length` (number of words in all sentences not removing duplicates) then maybe it can be assumed that `X_vec.shape[1] > max_sequence_length` can't occur?

Perhaps `sum([len(x) for x in X_buffer]) < max_sequence_length` never holds though.

Reply:
len(x) is the number of characters though. the expensive bit is tokenisation, going from text to tokens which you cannot avoid. the only thing that can go faster is padding which is not strictly necessary but it performs exactly the same operation I implement under the hood, maybe faster maybe slower.

note that transform is essentially a two lines function
```
sequences = self.tokenizer.encode(X)
return pad_sequences(sequences, maxlen=self.sequence_length)
```

I will investigate a bit the timing before I merge to see whether there is a cost saving I could do 😬

Reply:
So created this script

```
import json
import time

from tensorflow.keras.preprocessing.sequence import pad_sequences
import tensorflow as tf


DATA_PATH = "sample_disease_mesh.jsonl"


def yield_texts(DATA_PATH):
    with open(DATA_PATH) as f:
        for line in f:
            item = json.loads(line)
            yield item["text"]

tokenizer = tf.keras.preprocessing.text.Tokenizer()
tokenizer.fit_on_texts(yield_texts(DATA_PATH))

texts = list(yield_texts(DATA_PATH))
```
Version 1
```
start = time.time()
seqs = tokenizer.texts_to_sequences(texts)
pad_seqs = pad_sequences(seqs)
print(pad_seqs.shape[1])
print(time.time()-start)
```

Version 2
```
start = time.time()
seqs = tokenizer.texts_to_sequences(texts)
pad_seqs = pad_sequences(seqs)
print(pad_seqs.shape[1])
print(time.time()-start)
```

Results
* Version 1: 2.3s
* Version 2: 2s

So there is a slight speedup so will add the faster version in 🏎️

Reply:
glad youve found a speed up! v1 and v2 looks the same in this